esrally/mechanic/java_resolver.py

```diff
@@ -39,6 +39,9 @@ def determine_runtime_jdks():
 
     runtime_jdk_versions = determine_runtime_jdks()
     if runtime_jdk_versions[0] == "bundled":
+        if not allow_bundled_jvm:
+            raise exceptions.SystemSetupError("The bundled JDK is not allowed with the selected car(s): {}"
```

Comment:
This decision does not depend on the chosen car but rather on the Elasticsearch version. How about we change the error message to:

```
This Elasticsearch version does not contain a bundled JDK. Please specify a different runtime JDK.
```